types/react/index.d.ts

```diff
@@ -277,9 +277,9 @@ declare namespace React {
 
     // Base component for plain JS classes
     // tslint:disable-next-line:no-empty-interface
-    interface Component<P = {}, S = {}> extends ComponentLifecycle<P, S> { }
-    class Component<P, S> {
-        constructor(props: P, context?: any);
+    interface Component<P = {}, S = {}, DP extends Partial<P> = P> extends ComponentLifecycle<P, S> { }
```

Comment:
`P` and `S` were never really readable until you happened to know that was the order. `DP` is less readable. Thoughts on making them `<Props = {}, State = {}, DefaultProps extends Partial<Props> = Props>`? Or, if that's too big of a change for this, just the `DefaultProps extends Partial<Props> = Props`?

Reply:
Not in this PR for sure, but me and @MartynasZilinskas were talking about changing `P` into `Props` or `TProps` when big PRs will be merged (i.e. [new context API](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/24509)).